compiler/integration-tests/test/fuzzy/fuzzy.test.ts

```diff
@@ -0,0 +1,106 @@
+import { expect } from "@esm-bundle/chai";
+import { initializeResolver } from "@noir-lang/source-resolver";
+import newCompiler, { compile } from "@noir-lang/noir_wasm";
+import { acvm, abi } from "@noir-lang/noir_js";
+import { getRandomMap, getRandomString, getRandomUint8Array } from "./helper";
+import * as TOML from "smol-toml";
+
+const { default: initACVM, executeCircuit, compressWitness } = acvm;
+const { default: newABICoder, abiEncode } = abi;
+
+const iterations = 5;
+
+describe("compile fuzz test", () => {
+  for (let i = 0; i < iterations; i++) {
+    it("should error compile() gracefully", async () => {
+      await newCompiler();
+      await newABICoder();
+      await initACVM();
+      const fuzzedNoirSource = getRandomString(
+        Math.floor(Math.random() * Math.pow(10, i)),
+      );
+      initializeResolver(() => fuzzedNoirSource);
+      let compileOutput;
+      let errorOccurred = false;
+      try {
+        compileOutput = await compile({});
+        expect(compileOutput).to.be.an("object");
+      } catch (e) {
+        expect(e.toString().includes("RuntimeError: unreachable executed")).to
+          .be.true;
+        errorOccurred = true;
+      }
+      expect(
+        errorOccurred,
+        "Compilation should have failed with the fuzzed input",
+      ).to.be.true;
+    });
+  }
+});
+
+describe("abiEncode fuzz test", () => {
+  for (let i = 0; i < iterations; i++) {
+    it(`should error abiEncode() gracefully`, async () => {
+      let errorTriggered = false;
+      const stringLength = Math.pow(10, i);
+      const randomString = getRandomString(stringLength);
+      const fuzzedTOML = `${randomString} = "${randomString}"`;
+      try {
+        const inputs = TOML.parse(fuzzedTOML);
+        expect(inputs).to.be.an("object");
+        abiEncode(randomString, inputs, null);
+      } catch (e) {
+        console.log({ e });
+        expect(e.toString().includes("Error: invalid type")).to.be.true;
+        errorTriggered = true;
+      }
+
+      expect(errorTriggered).to.be.true;
+    });
+  }
+});
+
+// timeouts and errors in acvm_js
+xdescribe("executeCircuit fuzz test", function () {
+  this.timeout(120e3);
+
+  for (let i = 0; i < iterations; i++) {
+    it(`should error executeCircuit() gracefully`, async () => {
+      let errorTriggered = false;
+      const numberLength = Math.pow(10, i);
+      try {
+        const compressedByteCode = getRandomUint8Array(numberLength);
```

Comment:
Again, we'll always fail with the same error due to this not deserializing into valid ACIR. Why not just add a test where we pass in some invalid bytes once rather than multiple times?